opensourceleg/sensors/base.py

```diff
@@ -0,0 +1,164 @@
+﻿from abc import ABC, abstractmethod
+from functools import wraps
+
+import numpy as np
+
+
+class SensorNotStreamingException(Exception):
+    def __init__(self, sensor_name: str = "Sensor") -> None:
+        super().__init__(
+            f"{sensor_name} is not streaming, please ensure that the connections are intact, power is on, and the start method is called."
+        )
+
+
+def check_sensor_stream(func):
+    @wraps(func)
+    def wrapper(self, *args, **kwargs):
+        if not self.is_streaming:
+            raise SensorNotStreamingException(sensor_name=self.__repr__())
+        return func(self, *args, **kwargs)
+
+    return wrapper
+
+
+class SensorBase(ABC):
+    def __init__(self) -> None:
+        pass
+
+    def __repr__(self) -> str:
+        return f"SensorBase"
+
+    @abstractmethod
+    def start(self) -> None:
+        pass
+
+    @abstractmethod
+    def stop(self) -> None:
+        pass
+
+    @abstractmethod
+    def update(self) -> None:
+        pass
+
+    def __enter__(self):
+        self.start()
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        self.stop()
+
+    @property
+    @abstractmethod
+    def is_streaming(self) -> bool:
+        pass
+
+
+class EncoderBase(SensorBase, ABC):
```

Comment:
Purely for my learning here, why do we need to inherit from ABC if sensor base already inherits from it? Seems like double inheritance.